Review of the GSG - first part (#408)
neo4j/docs-getting-started: modules/ROOT/pages/whats-neo4j.adoc

=== Self-managed local deployment

If you prefer to work with a local deployment, you can follow the same instructions: install link:{docs-home}/desktop-manual[Neo4j Desktop] in case you are not working in a production environment or link:{docs-home}/operations-manual/current/installation/[install Neo4j] to your machine.

==== Neo4j on Docker

link:{docs-home}/operations-manual/current/docker[Docker] can be downloaded for macOS, Windows, and Linux from the link:https://www.docker.com/get-started[DockerHub].
It hosts an official Neo4j image that provides a standard, ready-to-run package of Neo4j Community Edition and Enterprise Edition for a variety of versions.

==== Neo4j on Kubernetes

With Neo4j Helm charts, you can deploy both a standalone and a cluster deployment of link:{docs-home}/operations-manual/current/kubernetes[Neo4j on Kubernetes], and use configuration options suitable for the most common scenarios. 

[NOTE]
Neo4j has free and subscription-based licensing options.
Read more about the link:https://neo4j.com/licensing/[available editions].

== Work with data

> After creating your database, you can take different paths depending on whether you want to work with your own data or use Neo4j's example datasets:

Reviewer: ```suggestion
After creating your database, your learning can take different paths depending on whether you want to work with your own data or use Neo4j's example datasets:
```

* *Own data*: There are several ways to link:{docs-home}/import[import] data to Neo4j.

> In case your dataset is not set for a graph database, it is recommended to link:{docs-home}/getting-started/data-modeling/[model] it for a better experience.

Reviewer: This is slightly problematic and I'm afraid we simplify too much. Their data has to be in the right format (csv or tsv I think are the only options at the moment) and if it's not, it can't be fixed by modeling it.

Author: Yes, that's part of the importing step, which is addressed in neo4j.com/docs/import. There you find ways to import data in other formats (like json, xml, relational to graph etc). But importing the data is not enough to have the dataset adjusted to be used in a graph database, thus the recommendation to model it. I think I can reframe this sentence a little bit to include these details though, if you think it's best

Reviewer: I think you should rephrase, there's a danger in simplifying too much.

* *Example datasets*: Both link:{docs-home}/aura[Aura] and link:{docs-home}/browser-manual/[Neo4j Browser] feature embedded guides that allow you to create example datasets and start querying.

> To access them, write `:guide` in Aura’s *Query* tab or in Neo4j Browser. +

Reviewer: Only Browser has the old embedded guides. Workspace has other guides, but they're accessible from the "Access all Neo4j Guides" button.

> * *Example datasets*: Both link:{docs-home}/aura[Aura] and link:{docs-home}/desktop-manual/[Neo4j Desktop] feature embedded guides that allow you to create example datasets and start querying.

Reviewer: ```suggestion
* *Example datasets*: Both link:{docs-home}/aura[Aura] and link:{docs-home}/browser-manual/[Neo4j Browser] feature embedded guides that allow you to create example datasets and start querying.
```

Author: That's indeed more correct, but we didn't mention Browser at all here, so I was wondering if it would be less confusing, though less accurate, to refer to Desktop instead of Browser?

Reviewer: Desktop doesn't have any embedded guide, so it would be wrong to say that, but Browser does. I don't think it's confusing, but you could add something like that you can open browser from Desktop if you think that would help.

Author: Yes, I think it's helpful because there are so many names of products 😅

> To access them, write `:guide` in Aura’s *Query* tab or in Browser, if you are using Neo4j Desktop. 

Reviewer: ```suggestion
To access them, write `:guide` in Aura Console's *Query*  or in Browser. 
```

You can use them to solve graph problems, such as predicting missing relationships, for example.

* The link:{docs-home}/ogm-manual[Object Graph Mapping (OGM)] library, maps nodes and relationships in the graph to objects and references in a domain model.
You can use this resource to start tracking changes and minimize necessary updates and transitive persistence (reading and updating neighborhoods of an object).

=== APIs

Neo4j currently offers three proprietary link:{docs-home}/create-applications/#_apis[APIs]:

* The link:{docs-home}/http-api[Neo4j HTTP API] allows you to execute a series of Cypher statements against a Neo4j instance through HTTP requests.
* The link:{docs-home}/cdc[Change Data Capture (CDC) API] allows you to capture and track changes to your database in real-time, as well as keep data sources up to date.
* The link:{docs-home}/query-apip[Neo4j Query API] allows you to develop client applications in languages not currently supported by Neo4j.

[NOTE]
====
At link:https://neo4j.com/labs/[Neo4j Labs], you can find experimental projects including APIs, libraries, and visualization tools.
====

== Keep learning

> To learn more about xref:graph-database.adoc[what is a graph database] and xref:reference/graphdb-concepts/index.adoc[the concepts] behind the technology, continue reading the documentation or check xref:reference/resources.adoc[other curated resources].

Reviewer: ```suggestion
To learn more about xref:graph-database.adoc[what a graph database is] and xref:reference/graphdb-concepts/index.adoc[the concepts] behind the technology, continue reading the documentation or browse xref:reference/resources.adoc[other curated resources].
```

> You can also reach out to other members of our community on the link:https://community.neo4j.com/c/neo4j-graph-platform/cypher/12?ref=guides[Neo4j Community Site].

Reviewer: ```suggestion
You can also reach out to other members of the Neo4j community on the link:https://community.neo4j.com/c/neo4j-graph-platform/cypher/12?ref=guides[Neo4j Community Site].
```

= What is Neo4j?
:description: An introduction on how to get started with Neo4j.
:!page-layout:
include::https://raw.githubusercontent.com/neo4j-graphacademy/courses/main/asciidoc/ads/generic.adoc[]

[.center]
image::ROOT:neo4j-architecture-diagram.svg[]

Neo4j is a _native graph database_, which means that it implements a true <<graph>> model all the way down to the storage level.
Instead of using a "graph abstraction" on top of another technology, the data is stored in Neo4j in the same way you may whiteboard your ideas.

Since 2007, Neo4j has evolved into a rich ecosystem of tools, applications, and libraries. 
This ecosystem allows you to integrate graph technologies with your working environment in a number of ways which are here described.

Beyond the core graph, Neo4j also provides ACID transactions, <<cluster>> support, and runtime failover.

[NOTE]
====
Neo4j is written in Java and Scala.
You can check the source code on https://github.com/neo4j/neo4j[GitHub^].
====

== How to interact with Neo4j

> Neo4j uses xref:cypher.adoc[Cypher], a declarative query language similar to SQL, but optimized for graphs.

Reviewer 2: ```suggestion
Neo4j uses xref:cypher.adoc[Cypher], a declarative, link:https://neo4j.com/docs/cypher-manual/current/appendix/gql-conformance/[GQL conformant] query language similar to SQL, but optimized for graphs.
```
